packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/invoke.ts

```diff
@@ -0,0 +1,277 @@
+  readonly authType?: sfn.TaskInput;
+
+}
+/**
+ * Invoke an API endpoint as a Task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-api-gateway.html
+ */
+export class ApiGatewayInvoke extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: ApiGatewayInvokeProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, ApiGatewayInvoke.SUPPORTED_INTEGRATION_PATTERNS);
+    const authType = this.props.authType ? this.props.authType.value : sfn.TaskInput.fromDataAt('$.AuthType').value;
+    if (authType === 'IAM_ROLE') {
+      const resource = props.apiEndpoint.split('.', 1)[0] + '/' + (props.stage ? props.stage + '/' : '$default/') + props.method + '/' + (props.path ?? '');
+
+      this.taskPolicies = [
+        new iam.PolicyStatement({
+          resources: [
+            Stack.of(this).formatArn({
+              service: 'execute-api',
+              resource: resource,
+            }),
+          ],
+          actions: ['execute-api:Invoke'],
+        }),
+      ];
+    } else if (authType === 'RESOURCE_POLICY') {
+      if (!sfn.FieldUtils.containsTaskToken(props.headers)) {
+        throw new Error('Task Token is required in `headers` Use JsonPath.taskToken to set the token.');
+      }
+      const resource = props.apiEndpoint.split('.', 1)[0] + '/' + (props.stage ? props.stage + '/' : '') + props.method + '/' + (props.path ? props.path + '/*' : '*');
+
+      this.taskPolicies = [
+        new iam.PolicyStatement({
+          resources: [
+            Stack.of(this).formatArn({
+              service: 'execute-api',
+              resource: resource,
+            }),
+          ],
+          actions: ['execute-api:Invoke'],
+          conditions: {
+            StringEquals: {
+              'aws:SourceArn': '*',
+            },
+          },
+        }),
+      ];
+    }
+  }
+
+  /**
+   * Provides the API Gateway Invoke service integration task configuration
+   */
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    if (this.props.headers && this.props.queryParameters && this.props.requestBody) {
+      return {
+        Resource: integrationResourceArn('apigateway', 'invoke', this.integrationPattern),
+        Parameters: sfn.FieldUtils.renderObject({
+          ApiEndpoint: this.props.apiEndpoint,
+          Method: this.props.method,
+          Headers: this.props.headers ? this.props.headers.value : sfn.TaskInput.fromDataAt('$.Headers').value,
```

Comment:
The same question as before (and at all the places). We have already checked for `this.props.header` in the `if` condition. Why do we need to check it again here? Also, where is this value coming from?

```ts
sfn.TaskInput.fromDataAt('$.Headers')
```